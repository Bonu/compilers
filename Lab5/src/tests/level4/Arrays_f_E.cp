class Main {

  int i;
  int j;

  int la[50];

  int ia1[50];
  int ia2[50];

  float fa1[50];
  float fa2[50];
  float a;
  float b;

  bool p;
  bool q;

  void main(){
	i = la[a];
	i = la[2.0];
	b = ia[3] + fa[i%j];
	i = ia[la[10]] * 3;
	i = la[i+j*a];
	i = la[fa1[2]];
	p = fa1[ia1[la[ia2[a]]]] != fa2[ia1[b]];
	a = fa1[ia1[2] * (i*j+i%j*2.1)];
  }
}
