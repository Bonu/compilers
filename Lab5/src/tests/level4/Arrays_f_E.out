line 20: invalid array index -- requires int or char type 
           but found ` a ' which has type float
line 21: invalid array index -- requires int or char type 
           but found ` 2.0 ' which has type float
line 22: unable to resolve name -- ia
line 22: unable to resolve name -- fa
line 23: unable to resolve name -- ia
line 24: invalid array index -- requires int or char type 
           but found ` i + j * a ' which has type float
line 25: invalid array index -- requires int or char type 
           but found ` fa1 [ 2 ] ' which has type float
line 26: invalid array index -- requires int or char type 
           but found ` a ' which has type float
line 26: invalid array index -- requires int or char type 
           but found ` b ' which has type float
line 27: invalid array index -- requires int or char type 
           but found ` ia1 [ 2 ] * ( i * j + i % j * 2.1 ) ' which has type float
10 errors were detected
